src/tool/hpcstruct/pmake.txt

```diff
@@ -0,0 +1,198 @@
+
+#-------------------------------------------------------------------------------
+# set up subdirectory paths
+#-------------------------------------------------------------------------------
+GPUBIN_DIR  = $(MEAS_DIR)/gpubins
+STRUCTS_DIR = $(MEAS_DIR)/structs
+
+#*******************************************************************************
+# a helper template makefile used by hpcstruct at runtime
+#
+# if hpcstruct is passed the name of a measurements directory that contains
+# a gpubins subdirectory, this makefile will be used to orchestrate parallel
+# analysis of all gpubins within the subdirectory.
+#
+# to simplify things at execution time, this makefile will be incorporated
+# into hpcstruct as a string and written to a temporary file if it is needed.
+# this avoids the need for hpcstruct to know how to find a copy of this
+# makefile at runtime in an hpctoolkit installation.
+#*******************************************************************************
+
+#*******************************************************************************
+#*******************************************************************************
+
+#*******************************************************************************
+# enable analysis of GPU binaries
+#*******************************************************************************
+ifeq ($(GPU_ANALYZE),1)
+
+#-------------------------------------------------------------------------------
+# $(GB): gpubin files
+#-------------------------------------------------------------------------------
+GB := $(wildcard $(GPUBIN_DIR)/*.gpubin)
+
+#-------------------------------------------------------------------------------
+# $(GS): hpcstruct files for gpubins
+#-------------------------------------------------------------------------------
+GS := $(patsubst $(GPUBIN_DIR)/%,$(STRUCTS_DIR)/%.hpcstruct,$(GB))
+
+#-------------------------------------------------------------------------------
+# $(GW): warning files that may be generated during structure analysis of gpubins
+#-------------------------------------------------------------------------------
+GW := $(patsubst %.hpcstruct,%.warnings,$(GS))
+
+endif
+
+#*******************************************************************************
+#*******************************************************************************
+
+#*******************************************************************************
+# enable analysis of CPU binaries
+#*******************************************************************************
+ifeq ($(CPU_ANALYZE),1)
+CPUBIN_DIR  = $(MEAS_DIR)/cpubins
+LM_DIR      = $(MEAS_DIR)/lm
+
+#-------------------------------------------------------------------------------
+# $(H): hpcrun file for thread 0 in each process
+#-------------------------------------------------------------------------------
+H := $(wildcard $(MEAS_DIR)/*-000000-000-*.hpcrun)
```

Comment:
@jmellorcrummey What if there's no thread 0 in this measurement dir or thread 0 does not contain all necessary load modules?

Reply:
When the measurement library is initialized, there is always a thread 0. It is possible (but unlikely) that thread 0 will finish early and only record a prefix of the roadmap. I decided to choose thread 0 rather than processing hpcrun files for all threads, which will typically be duplicates of thread 0. I think that this will do the right thing in all but exceptional cases. 

We could add an option to consider all threads.

Reply:
Thanks for the explanation!

> We could add an option to consider all threads.

That makes sense.

Reply:
@Jokeren After further consideration, I decided to analyze all hpcrun files all the time. Scraping load modules from hpcrun files should be much faster than running hpcstruct on anything, so I don't think it will be a bottleneck. 

Other changes: 
- Don't remove files with listing of load modules per hpcrun file. this improves incrementality if it is interrupted and restarted. 
- Ensure that pushd output is suppressed.